components/TransactionForm.js

```diff
@@ -55,13 +63,15 @@ export default function TransactionForm({
     }
 
     onSubmit({ ...initialData, ...data });
+
     event.target.reset();
     setAmount("");
     setSelectedCategoryInForm("");
   }
 
-  function handleCategorySelect(categoryName) {
-    setSelectedCategoryInForm(categoryName);
+  function handleCategorySelect(category) {
+    setSelectedCategoryInForm(category.name);
+    setSelectedCategory(category._id);
```

Comment:
why do you need both states? You could use only set SelectedCategory and use `categories.find()` to get the remaining information with the `_id`.

Reply:
Thanks for the tip, we decided to do this in our refactoring PR in last Sprint.